app/models/user.rb

```diff
@@ -4,13 +4,21 @@ class User < ApplicationRecord
   validates :email, uniqueness: true
   validates :role, presence: true
   validate :password_regex
-  has_many :profiles
+  has_one :profile, dependent: :destroy
+  has_many :pathways, dependent: :destroy
```

Comment:
Should this be has_many :user_pathways?

Reply:
As per the schema design, yes. A user should be associated to a pathway through a user_pathway.
Or if the user is the creator of a pathway, should be directly related to the pathway. (I.e pathway.user returns the creator of the pathway)

I only added this line ` has_many :pathways, dependent: :destroy` because there was an incomplete association in our database between pathway and user (`app/models/pathway.rb` pathway had a belongs_to association to user model, but user model didn't has_many pathways) so we were unable to destroy a user that had pathways.

I think we may want to:
- add association so mentee user has_many pathways through user_pathways
- rename if possible the direct association between pathway and user, so we have a creator_id field in pathways which contains the user foreign key. There wouldn't be a big difference, just less confusing
But these should be done in separate tasks

Reply:
lets chat tomorrow about this association